src/commands/kv/list_keys.rs

```diff
@@ -0,0 +1,82 @@
+extern crate serde_json;
+
+use cloudflare::endpoints::workerskv::list_namespace_keys::ListNamespaceKeys;
+use cloudflare::endpoints::workerskv::list_namespace_keys::ListNamespaceKeysParams;
+use cloudflare::framework::apiclient::ApiClient;
+use failure::bail;
+use serde_json::value::Value as JsonValue;
+
+// Note: this function only prints keys in json form, given that
+// the number of entries in each json blob is variable (so csv and tsv
+// representation won't make sense)
+pub fn list_keys(id: &str, prefix: Option<&str>) -> Result<(), failure::Error> {
+    let client = super::api_client()?;
+    let account_id = super::account_id()?;
+
+    let params = ListNamespaceKeysParams {
+        limit: None, // Defaults to 1000 (the maximum)
+        cursor: None,
+        prefix: prefix.map(str::to_string),
+    };
+
+    let mut request_params = ListNamespaceKeys {
+        account_identifier: &account_id,
+        namespace_identifier: id,
+        params: params,
+    };
+
+    let mut response = client.request(&request_params);
+
+    print!("["); // Open json list bracket
+
+    // Iterate over all pages until no pages of keys are left.
+    // This is detected when a returned cursor is an empty string.
+    // todo(gabbi): the code in this loop is the product of a looooong fight
+    // with the borrow checker. Please tell me if there's a neater way to write
+    // the logic below!
+    loop {
+        let (result, cursor) = match response {
+            Ok(ref success) => (
+                serde_json::to_string(&success.result.clone())?,
```

Comment:
rather than deserializing here and then slicing the result in `print_page`, perhaps let it be the `Vec` and iterate over that to print the objects?

Reply:
I thought about doing just that :smile:! I wondered if it would be slower to iterate through the vec and serialize each entry instead of printing out a few serialized "batches", so to speak. I'd love to defer to @ashleygwilliams or @steveklabnik (someone who knows more about rust than me) to see which approach seems more efficient.

Reply:
if i put on my dubs hat, i would be willing to bet that the answer would be somewhere along these lines:
The performance degradation from such a change would pale in comparison to the http requests we are already making in between prints. go for max readability and we can adjust from there if we find performance issues.

Reply:
also, if i'm understanding the code right, since we are using cloudflare rs, once we make the request, we are already bringing in the json and deserializing it to a `Vec<Key>` before this step, only to re-serialize it into a string and then slice it, which i believe makes a copy of the String. I think that all comes out to 6/half dozen compared to deserializing each Key and just printing it. This is my conjecture anyway.

Reply:
Benchmarked, and many individual serializes are not noticeably different from a few large serializes. We'll go with many individual serializes for readability and easier output formatting :)